File: c++/CountAndSay.cpp
class Solution
{
  public:
    string countAndSay(int n)
    {
        if (n == 0)
            return 0;

        string res = "1";
        while (--n)
        {
            string seq = "";
            int count = 0;
            char prev = res[0];
            for (int i = 0; i < res.size(); i++)
            {
                if (res[i] == prev)
                {
                    count++;
                }
                if (res[i] != prev)
                {
                    seq += to_string(count) + prev;
                    count = 1;
                }

                prev = res[i];
            }
            seq += to_string(count) + prev;
            res = seq;
        }
        return res;
    }
};

File: python/count_and_say.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        newres = '1'
        for j in range(n):
            res = newres
            count = 0
            prev = res[0]
            newres = ''
            size = len(res)
            for i in range(size):
                if res[i] == prev:
                    count+=1
                else:
                    newres += str(count) + prev
                    count = 1
                if (i == size - 1):
                    newres += str(count) + res[i]
                prev = res[i]

        return res